CovertInfiltration/Src/CovertInfiltration/Classes/UISS_InfiltrationPanel.uc

```diff
@@ -0,0 +1,128 @@
+	SquadDurationDisplay = Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, 23, 247);
+
+	TotalDurationTitle = Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, -29, 73);
+	TotalDurationTitle.SetSubTitle(default.strTotalDurationTitle, "FAF0C8");
+
+	BaseDurationTitle = Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, 20, 153);
+	BaseDurationTitle.SetSubTitle(default.strBaseDurationTitle);
+
+	SquadDurationTitle = Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, 20, 219);
+	SquadDurationTitle.SetSubTitle(default.strSquadDurationTitle);
+
+	RisksTitle = Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, 20, 285);
+	RisksTitle.SetSubTitle(class'UICovertActions'.default.CovertActions_RiskTitle, "FAF0C8");
+
+	RiskLabelsOffsetY = 32;
+	RiskLabels.AddItem(Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, 23, 316));
+}
+
+simulated function UpdateData(XComGameState_CovertAction CurrentAction)
+{	
+	local XComGameState_HeadquartersXCom XComHQ;
+	local int BaseDuration, SquadDuration;
+
+	XComHQ = class'UIUtilities_Strategy'.static.GetXComHQ();
+
+	BaseDuration = CurrentAction.HoursToComplete;
+
+	SquadDuration = class'X2Helper_Infiltration'.static.GetSquadInfiltration(XComHQ.Squad);
+
+	TotalDurationDisplay.SetInfoValue(GetDaysAndHoursString(BaseDuration + SquadDuration), class'UIUtilities_Colors'.const.NORMAL_HTML_COLOR);
+	BaseDurationDisplay.SetInfoValue(GetDaysAndHoursString(BaseDuration), class'UIUtilities_Colors'.const.NORMAL_HTML_COLOR);
+	SquadDurationDisplay.SetInfoValue(GetDaysAndHoursString(SquadDuration, default.strPlusDaysAndHours), class'UIUtilities_Colors'.const.NORMAL_HTML_COLOR);
+
+	UpdateRiskLabels(CurrentAction);
+}
+
+simulated function UpdateRiskLabels(XComGameState_CovertAction CurrentAction)
+{
+	local array<string> RiskStrings;
+	local int idx;
+
+	RiskStrings = class'UIUtilities_Infiltration'.static.GetRisksStringsFor(CurrentAction);
+
+	for (idx = 0; idx < RiskStrings.Length; idx++)
+	{
+		GetRiskLabel(idx).SetText(RiskStrings[idx]);
+	}
+}
+
+static function string GetDaysAndHoursString(int iHours, optional string locString)
+{
+	local int ActualHours, ActualDays;
+	local XGParamTag ParamTag;
+	local string ReturnString;
+
+	if(locString == "")
+		locString = default.strDaysAndHours;
+
+	ActualDays = iHours / 24;
+	ActualHours = iHours % 24;
+
+	ParamTag = XGParamTag(`XEXPANDCONTEXT.FindTag("XGParam"));
+	ParamTag.IntValue0 = ActualDays;
+	ParamTag.IntValue1 = ActualHours;
+	ReturnString = `XEXPAND.ExpandString(locString);
+
+	return ReturnString;
+}
+
+simulated function UISS_InfiltrationItem GetRiskLabel(int index)
+{
+	local UISS_InfiltrationItem newLabel;
+
+	if (index < RiskLabels.Length)
+		return RiskLabels[index];
+
+	newLabel = Spawn(class'UISS_InfiltrationItem', self).InitObjectiveListItem(, RiskLabels[RiskLabels.Length - 1].X, RiskLabels[RiskLabels.Length - 1].Y + RiskLabelsOffsetY);
+
+	RiskLabels.AddItem(newLabel);
```

Comment:
This assumes that `index` is within `[0; RiskLabels.Length]`. The only time you call this here, this precondition is met, but it may be worth leaving a comment about it here.